solr-http-retry.go

```diff
@@ -0,0 +1,80 @@
+package solr
+
+import (
+	"log"
+	"net/url"
+	"time"
+)
+
+type SolrHttpRetrier struct {
+	solrCli            SolrHTTP
+	retries            int
+	exponentialBackoff time.Duration
+	timeout            time.Duration
+}
+
+func NewSolrHttpRetrier(solrHttp SolrHTTP, retries int, exponentialBackoff time.Duration, timeout time.Duration) SolrHTTP {
+
+	solrRetrier := SolrHttpRetrier{solrCli: solrHttp, retries: retries, exponentialBackoff: exponentialBackoff, timeout: timeout}
+	return &solrRetrier
+}
+
+func (s *SolrHttpRetrier) Read(opts ...func(url.Values)) (SolrResponse, error) {
+	now := time.Now()
+	var resp SolrResponse
+	var err error
+	for attempt := 0; attempt < s.retries; attempt++ {
+		resp, err = s.solrCli.Read(opts...)
+		if err != nil {
+			s.Logger().Printf("[Solr Http Retrier] Error Retrying %v ", err)
+			if !s.backoff(now, attempt) {
+				s.Logger().Println("[Solr Http Retrier] Timeout exceeded", err)
+				break
+			}
+			continue
+		}
+		break
+	}
+	return resp, err
+}
+
+func (s *SolrHttpRetrier) Update(docID string, updateOnly bool, doc interface{}, opts ...func(url.Values)) error {
+	now := time.Now()
+	var err error
+	for attempt := 0; attempt < s.retries; attempt++ {
```

Comment:
nitpick: perhaps you could pass in a callback to a `retry` func that has this retry logic.  This callback could be the solrClient update or read.